x-pack/plugins/security_solution/public/flyout/document_details/right/hooks/use_show_event_overview.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { EcsSecurityExtension as Ecs } from '@kbn/securitysolution-ecs';
+import { getField, getFieldArray } from '../../shared/utils';
+import type { GetFieldsData } from '../../../../common/hooks/use_get_fields_data';
+import { getRowRenderer } from '../../../../timelines/components/timeline/body/renderers/get_row_renderer';
+import { defaultRowRenderers } from '../../../../timelines/components/timeline/body/renderers';
+import { isEcsAllowedValue } from '../utils/event_utils';
+
+export interface UseShowEventOverviewParams {
+  /**
+   * Retrieves searchHit values for the provided field
+   */
+  getFieldsData: GetFieldsData;
+  /**
+   * An object with top level fields from the ECS object
+   */
+  dataAsNestedObject: Ecs;
+}
+
+/**
+ * Hook to return true if overview should be visible
```

Comment:
sorry for being annoying here, but let's improve this description a bit, a lot of people not familiar with the codebase wouldn't understand what this does.
Something like
```
Hook used on the right panel to decide if the overview section should be displayed on not. This also helps deciding if the flyout has an expanded section.
The hook looks at the `event.kind` and `event.category` fields of the document.
```

Reply:
actually, something feel off about this hook or where it's used.

I see this in the parent

```
const isEventKindSignal = getField(getFieldsData('event.kind')) === EventKind.signal;const expandableEventFlyoutEnabled = useIsExperimentalFeatureEnabled('expandableEventFlyoutEnabled');
const showEventOverview =
 useShowEventOverview({ getFieldsData, dataAsNestedObject }) && expandableEventFlyoutEnabled;
```

then in the `useMemo` just below that we do
```
return isEventKindSignal || showEventOverview
{...}
```

it feels a it odd to name this hook `useShowEventOverview` while we only use it for non-signal documents.
I was about to suggest renaming to something else, but I can't find a name that works because it actually does a lot.
So my suggestion is why not adding more to this hook, by incorporating
```
const isEventKindSignal = getField(getFieldsData('event.kind')) === EventKind.signal;
```
and do the whole check here.

We could push this even further and have it return the list of tabs. Therefore we could rename the hook `useRenderTabs` (or something better).

Wdyt?

Reply:
Well I see that you're using the result here as well
```
<PanelNavigation flyoutIsExpandable={isEventKindSignal || showEventOverview} />
```
so maybe drop the idea of returning the panels, but we could still have the hook renamed to something else. It feels weird to say that the flyout is expandable if event kind is signal or if we show overview...

Reply:
@PhilippeOberti thanks for the thorough review! I played around with it and moved almost all the logic to the hook `useFlyoutIsExpandable`. This also remove all the signal/non-signal checks and made it cleaner for those unfamiliar with the logic.